test/generator/exercise_test.rb

```diff
@@ -83,21 +88,27 @@ def test_bookkeeping
   end
 end
 TESTS_FILE
-      mock_file = Minitest::Mock.new.expect :write, expected_content.length, [expected_content]
-      subject = Exercise.new(paths: FixturePaths, slug: 'alpha')
-      GitCommand.stub(:abbreviated_commit_hash, '123456789') do
-        File.stub(:open, true, mock_file) do
-          assert_equal expected_content, subject.build_tests
-        end
-      end
-      mock_file.verify
-      # Don't pollute the namespace
-      Object.send(:remove_const, :AlphaCase)
-    end
 
-    def test_name
-      subject = Exercise.new(paths: FixturePaths, slug: 'alpha-beta')
-      assert_equal 'alpha_beta', subject.name
+     mock_repository = Minitest::Mock.new
+     mock_template_values = Minitest::Mock.new
+
+     mock_template_file = Minitest::Mock.new.expect :to_s, nil
+     mock_repository.expect :tests_template, mock_template_file
+
+     subject = Exercise.new(repository: mock_repository)
+     subject.stub :template_values, mock_template_values do
+
+       mock_minitest_file = Minitest::Mock.new.expect(:generate, expected_content,
+         [template: mock_template_file, values: mock_template_values])
+
+       mock_repository.expect :minitest_tests, mock_minitest_file
+
+       assert_equal expected_content, subject.build_tests
+
+       mock_repository.verify
+       mock_minitest_file.verify
+       mock_template_file.verify
```

Comment:
There are too many things being asserted/verified in all the tests that have changed in this commit.
Refactor later.